kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaCreateForm.java

```diff
@@ -6,73 +6,78 @@
+    protected SelenideElement loadingSpinner = $x("//*[contains(text(),'Loading')]");
+    protected SelenideElement schemaNameField = $x("//input[@name='subject']");
+    protected SelenideElement pageTitle = $x("//h1['Edit']");
+    protected SelenideElement schemaTextArea = $x("//textarea[@name='schema']");
+    protected SelenideElement submitBtn = $x("//button[@type='submit']");
+    protected SelenideElement newSchemaInput = $("#newSchema [wrap]");
+    protected SelenideElement schemaTypeDdl = $x("//ul[@name='schemaType']");
+    protected SelenideElement compatibilityLevelList = $x("//ul[@name='compatibilityLevel']");
+    protected SelenideElement newSchemaTextArea = $x("//div[@id='newSchema']");
+    protected String elementLocatorDdl = "//li[text()='%s']";
 
     @Step
-    public SchemaCreateForm waitUntilScreenReady() {
-        $x("//h1['Edit']").shouldBe(Condition.visible);
+    public SchemaCreateForm waitUntilScreenReady(){
+        loadingSpinner.shouldBe(Condition.disappear);
+        pageTitle.shouldBe(Condition.visible);
         return this;
     }
 
     @Step
     public SchemaCreateForm setSubjectName(String name) {
-        subjectName.setValue(name);
+        schemaNameField.setValue(name);
         return this;
     }
 
     @Step
     public SchemaCreateForm setSchemaField(String text) {
-        schemaField.setValue(text);
+        schemaTextArea.setValue(text);
         return this;
     }
 
     @Step
     public SchemaCreateForm selectSchemaTypeFromDropdown(SchemaType schemaType) {
-        $("ul[role='listbox']").click();
-        $x("//li[text()='" + schemaType.getValue() + "']").click();
+        schemaTypeDdl.shouldBe(Condition.enabled).click();
+        $x(String.format(elementLocatorDdl, schemaType.getValue())).shouldBe(Condition.visible).click();
         return this;
     }
 
     @Step
-    public SchemaDetails clickSubmit() {
-        clickByJavaScript(submitSchemaButton);
-        return new SchemaDetails();
+    public SchemaCreateForm clickSubmitBtn() {
+        clickByJavaScript(submitBtn);
+        return this;
     }
 
     @Step
     public SchemaCreateForm selectCompatibilityLevelFromDropdown(CompatibilityLevel.CompatibilityEnum level) {
-        $x("//ul[@name='compatibilityLevel']").click();
-        $x("//li[text()='" + level.getValue() + "']").click();
+        compatibilityLevelList.shouldBe(Condition.enabled).click();
+        $x(String.format(elementLocatorDdl, level.getValue())).shouldBe(Condition.visible).click();
         return this;
     }
 
     @Step("Set new schema value")
     public SchemaCreateForm setNewSchemaValue(String configJson) {
-        $("#newSchema").click();
-        newSchemaTextArea.sendKeys(Keys.CONTROL + "a", Keys.BACK_SPACE);
+        newSchemaTextArea.shouldBe(Condition.visible).click();
+        clearByKeyboard(newSchemaInput);
         Selenide.executeJavaScript("arguments[0].value = '';", $("#newSchema"));
```

Comment:
can we simplify someway here?

Reply:
simplify with clickByJavaScript() method

Reply:
what??

Reply:
Deleted clickByJavaScript() , cause we cleared field in clearByKeyboard. JS injection here impractical